### PR TITLE
Kotlin migration ModelTest

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Kotlin Migration of ModelTest

### CODE REVIEW

**Comment by Prince-kushwaha** [comment]
@david-allison how i fix below unit test error if i not made change in UpgradeGesturesToControlsTest.kt
Screenshot 2022-06-25 213851

**Comment by mikehardy** [comment]
@Prince-kushwaha that failure is a flaky test - it's actually our most common flake at the moment, pretty irritating - hopefully either someone has time to fix it soon or I'll have to disable it in CI https://github.com/ankidroid/Anki-Android/issues/11091#issuecomment-1130226317

**Review by lukstbit — CHANGES_REQUESTED** [review]
Has conflicts at the moment.

File: AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.kt [hunk 1]
```diff
@@ -13,408 +13,504 @@
...
+import org.hamcrest.Matchers.endsWith
+import org.hamcrest.Matchers.not
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import java.util.*
+
+@RunWith(AndroidJUnit4::class)
+class ModelTest : RobolectricTest() {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.kt:40]: Add two cleanups:
```
@KotlinCleanup("fix IDE lint issues")
@KotlinCleanup("improve kotlin code where possible")
```

File: AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.kt [hunk 2]
```diff
@@ -424,280 +520,360 @@ public class ModelTest extends RobolectricTest {
...
-        assertListEquals(Arrays.asList(0, 1), Models.availOrds(reverse, new String[]{"Foo", "Bar"}, Models.AllowEmpty.TRUE));
-        assertListEquals(Arrays.asList(1), Models.availOrds(reverse, new String[]{"  \t ", "Bar"}, Models.AllowEmpty.TRUE));
-    }
-
-    /**
-     * tests if Model.getDid() returns model did
-     * or default deck id (1) if null
-     */
-    @Test
-    public void getDid_test() {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.kt]: Why did you delete this test method?

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.kt (hunk 1 above)
Comment (Prince-kushwaha) [reply AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.kt:41]: ok

**Review by lukstbit — APPROVED** [review]

**Comment by mikehardy** [comment]
Oh no, @Prince-kushwaha sorry about this - it appears to have taken a conflict before I had any time to work through the second review queue and get things merged. I'm really sorry about that. Do you have time to fix the conflict?

**Comment by Prince-kushwaha** [comment]
@mikehardy i fixed the conflict

**Review by Arthur-Milchior — APPROVED** [review]

**Pull request merged**